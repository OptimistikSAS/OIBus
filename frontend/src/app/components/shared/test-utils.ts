import { Provider } from '@angular/core';
import { NgbConfig } from '@ng-bootstrap/ng-bootstrap';

const NO_ANIMATION_NGB_CONFIG: NgbConfig = { animation: false };

export const noAnimation: Provider = { provide: NgbConfig, useValue: NO_ANIMATION_NGB_CONFIG };
