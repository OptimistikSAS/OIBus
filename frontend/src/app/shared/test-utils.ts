import { Provider } from '@angular/core';
import { NgbConfig } from '@ng-bootstrap/ng-bootstrap';
import { Page } from './types';

const NO_ANIMATION_NGB_CONFIG: NgbConfig = { animation: false };

export const noAnimation: Provider = { provide: NgbConfig, useValue: NO_ANIMATION_NGB_CONFIG };

export function toPage<T>(content: Array<T>, totalElements: number = content.length, num = 0, size = Math.max(20, totalElements)): Page<T> {
  return {
    content,
    totalElements,
    number: num,
    size,
    totalPages: Math.ceil(totalElements / size)
  };
}

export function emptyPage<T>(size = 20): Page<T> {
  return {
    totalElements: 0,
    content: [] as Array<T>,
    size,
    totalPages: 0,
    number: 0
  };
}
