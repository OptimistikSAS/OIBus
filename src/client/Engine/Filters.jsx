import React from 'react'
import PropTypes from 'prop-types'
import { Row, Col } from 'reactstrap'
import Table from '../components/table/Table.jsx'
import { OIbText, OIbTitle } from '../components/OIbForm'
import { ConfigContext } from '../context/configContext.jsx'
import validation from './Engine.validation'

const Filters = ({ filters }) => {
  const { dispatchNewConfig } = React.useContext(ConfigContext)
  const handleDelete = (rowIndex) => {
    dispatchNewConfig({ type: 'deleteRow', name: `engine.filter.${rowIndex}` })
  }
  const handleAdd = () => {
    dispatchNewConfig({ type: 'addRow', name: 'engine.filter', value: '' })
  }
  const onChange = (name, value, validity) => {
    dispatchNewConfig({ type: 'update', name, value, validity })
  }
  return (
    filters && (
      <>
        <OIbTitle label="IP Filters">
          <p>
            This is the list of IP adresses (or hostnames) that will be allowed to access the admin console.
            By default, it is only the local host. It is possible to use wildcards (such as 192.168.1.* or just
            * to allow every IP address) in the IP address to authorize a subnet for example.
          </p>
        </OIbTitle>
        <Row>
          <Col md={4}>
            <Table
              headers={['adresse']}
              rows={filters.map((filter, i) => [
                {
                  name: `engine.filter.${i}`,
                  value: (
                    <OIbText
                      name={`engine.filter.${i}`}
                      value={filter}
                      valid={validation.engine.filter}
                      onChange={onChange}
                      defaultValue=""
                    />
                  ),
                },
              ])}
              handleDelete={handleDelete}
              handleAdd={handleAdd}
            />
          </Col>
        </Row>
      </>
    )
  )
}

Filters.propTypes = { filters: PropTypes.arrayOf(String).isRequired }

export default Filters
