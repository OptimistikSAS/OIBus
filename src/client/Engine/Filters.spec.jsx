import React from 'react'
import ReactDOM from 'react-dom'
import { act } from 'react-dom/test-utils'

import testConfig from '../../../tests/testConfig'
import Filters from './Filters.jsx'

const mockMath = Object.create(global.Math)
mockMath.random = () => 1
global.Math = mockMath

let container
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  document.body.removeChild(container)
  container = null
})

describe('Filters', () => {
  test('check Filters', () => {
    act(() => {
      ReactDOM.render(<Filters
        filters={testConfig.engine.filter}
      />, container)
    })
    expect(container).toMatchSnapshot()
  })
})
