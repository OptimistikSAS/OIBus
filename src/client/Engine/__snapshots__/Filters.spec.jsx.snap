// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters check Filters 1`] = `
<div>
  <div
    class="row"
  >
    <h5>
      IP Filters
      <button
        aria-label="Close"
        class="close"
        id="id"
      >
        <svg
          class="oi-help"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 448c-110.532 0-200-89.431-200-200 0-110.495 89.472-200 200-200 110.491 0 200 89.471 200 200 0 110.53-89.431 200-200 200zm107.244-255.2c0 67.052-72.421 68.084-72.421 92.863V300c0 6.627-5.373 12-12 12h-45.647c-6.627 0-12-5.373-12-12v-8.659c0-35.745 27.1-50.034 47.579-61.516 17.561-9.845 28.324-16.541 28.324-29.579 0-17.246-21.999-28.693-39.784-28.693-23.189 0-33.894 10.977-48.942 29.969-4.057 5.12-11.46 6.071-16.666 2.124l-27.824-21.098c-5.107-3.872-6.251-11.066-2.644-16.363C184.846 131.491 214.94 112 261.794 112c49.071 0 101.45 38.304 101.45 88.8zM298 368c0 23.159-18.841 42-42 42s-42-18.841-42-42 18.841-42 42-42 42 18.841 42 42z"
          />
        </svg>
      </button>
    </h5>
  </div>
  <div
    class="container-fluid"
  >
    <div
      class="collapse"
      toggler="id"
    >
      <div
        class="row"
        style="margin-bottom: 15px;"
      >
        <div
          class="col"
        >
          <p>
            This is the list of IP adresses (or hostnames) that will be allowed to access the admin console. By default, it is only the local host. It is possible to use wildcards (such as 192.168.1.* or just * to allow every IP address) in the IP address to authorize a subnet for example.
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col-md-4"
    >
      <table
        class="table table-sm table-striped table-hover"
      >
        <thead>
          <tr>
            <th
              scope="col"
            >
              <svg
                class="oi-icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm144 276c0 6.6-5.4 12-12 12h-92v92c0 6.6-5.4 12-12 12h-56c-6.6 0-12-5.4-12-12v-92h-92c-6.6 0-12-5.4-12-12v-56c0-6.6 5.4-12 12-12h92v-92c0-6.6 5.4-12 12-12h56c6.6 0 12 5.4 12 12v92h92c6.6 0 12 5.4 12 12v56z"
                />
              </svg>
              <span>
                adresse
              </span>
            </th>
            <th
              scope="col"
            />
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              <div
                class="form-group"
                style="margin-bottom: 0px;"
              >
                <input
                  class="oi-form-input form-control"
                  id="engine.filter.0"
                  name="engine.filter.0"
                  type="text"
                  value="127.0.0.1"
                />
                <div
                  class="invalid-feedback"
                />
              </div>
            </td>
            <td>
              <svg
                class="oi-icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
                />
              </svg>
            </td>
          </tr>
          <tr>
            <td>
              <div
                class="form-group"
                style="margin-bottom: 0px;"
              >
                <input
                  class="oi-form-input form-control"
                  id="engine.filter.1"
                  name="engine.filter.1"
                  type="text"
                  value="::1"
                />
                <div
                  class="invalid-feedback"
                />
              </div>
            </td>
            <td>
              <svg
                class="oi-icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
                />
              </svg>
            </td>
          </tr>
          <tr>
            <td>
              <div
                class="form-group"
                style="margin-bottom: 0px;"
              >
                <input
                  class="oi-form-input form-control"
                  id="engine.filter.2"
                  name="engine.filter.2"
                  type="text"
                  value="::ffff:127.0.0.1"
                />
                <div
                  class="invalid-feedback"
                />
              </div>
            </td>
            <td>
              <svg
                class="oi-icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
                />
              </svg>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
